Sources/NIO/SelectableEventLoop.swift

```diff
@@ -104,6 +105,26 @@ internal final class SelectableEventLoop: EventLoop {
         }
     }
 
+    @usableFromInline
+    internal var _validInternalStateToScheduleTasks: Bool {
+        switch self.internalState {
+        case .exitingThread:
+            return false
+        default:
```

Comment:
could you change this from `default:` to enumerating the cases explicitly like `case .running..., .running..., .noLongerRunning`? It's important that we get the compiler errors if we add a case.